docs/getting-started-guides/kubeadm.md

```diff
@@ -54,358 +65,556 @@ If you are not constrained, there are some other tools built to give you complet
+kubeadm to tell it what to do.
+
+Note: Disabling SELinux by running `setenforce 0` is required in order to allow
+containers to access the host filesystem, which is required by pod networks for
+example. You have to do this until SELinux support is improved in the kubelet.
 
 ### (2/4) Initializing your master
 
-The master is the machine where the "control plane" components run, including `etcd` (the cluster database) and the API server (which the `kubectl` CLI communicates with).
-All of these components run in pods started by `kubelet` and the following images are required and will be automatically pulled by `kubelet` if they are absent while `kubeadm init` is initializing your master:
-
-    gcr.io/google_containers/kube-proxy-amd64                v1.5.3         
-    gcr.io/google_containers/kube-controller-manager-amd64   v1.5.3         
-    gcr.io/google_containers/kube-scheduler-amd64            v1.5.3         
-    gcr.io/google_containers/kube-apiserver-amd64            v1.5.3         
-    gcr.io/google_containers/etcd-amd64                      3.0.14-kubeadm 
-    gcr.io/google_containers/kube-discovery-amd64            1.0            
-    gcr.io/google_containers/pause-amd64                     3.0 
+The master is the machine where the "control plane" components run, including
+etcd (the cluster database) and the API server (which the kubectl CLI
+communicates with). All of these components run in pods started by kubelet and
+the following images are required and will be automatically pulled by kubelet
+if they are absent while `kubeadm init` is initializing your master:
+
+| Image Name | Version |
+|---|---|
+| gcr.io/google_containers/kube-apiserver-amd64 | v1.6.0
+| gcr.io/google_containers/kube-controller-manager-amd64 | v1.6.0
+| gcr.io/google_containers/kube-scheduler-amd64 | v1.6.0
+| gcr.io/google_containers/kube-proxy-amd64 | v1.6.0
+| gcr.io/google_containers/etcd-amd64 | 3.0.17
+| gcr.io/google_containers/pause-amd64 | 3.0
+| gcr.io/google_containers/k8s-dns-sidecar-amd64 | 1.14.1
+| gcr.io/google_containers/k8s-dns-kube-dns-amd64 | 1.14.1
+| gcr.io/google_containers/k8s-dns-dnsmasq-nanny-amd64 | 1.14.1
+
+To initialize the master, pick one of the machines you previously installed
+kubelet and kubeadm on, and run:
+
+``` bash
+kubeadm init
+```
 
-Right now you can't run `kubeadm init` twice without tearing down the cluster in between, see [Tear down](#tear-down).
+**Note:** this will autodetect the network interface to advertise the master on
+as the interface with the default gateway. If you want to use a different
+interface, specify `--api-advertise-addresses <ip-address>` argument to `kubeadm
+init`.
 
-If you try to run `kubeadm init` and your machine is in a state that is incompatible with starting a Kubernetes cluster, `kubeadm` will warn you about things that might not work or it will error out for unsatisfied mandatory requirements.
+When using [flannel](https://github.com/coreos/flannel) as the [pod
```

Comment:
@jbeda We've got quite many bug reports where kube-proxy can't distinguish internal/external communication:
```
kubectl -n kube-system logs kube-proxy-tkphx
I0315 18:39:01.406692       1 server.go:225] Using iptables Proxier.
W0315 18:39:03.015835       1 proxier.go:307] clusterCIDR not specified, unable to distinguish between internal and external traffic
I0315 18:39:03.015868       1 server.go:249] Tearing down userspace rules.
I0315 18:39:03.317259       1 conntrack.go:81] Set sysctl 'net/netfilter/nf_conntrack_max' to 131072
I0315 18:39:03.317680       1 conntrack.go:66] Setting conntrack hashsize to 32768
I0315 18:39:03.317922       1 conntrack.go:81] Set sysctl 'net/netfilter/nf_conntrack_tcp_timeout_established' to 86400
I0315 18:39:03.317955       1 conntrack.go:81] Set sysctl 'net/netfilter/nf_conntrack_tcp_timeout_close_wait' to 3600
```
this is without the cluster-cidr flag to kube-proxy